### PR TITLE
feature/convert text file#147

### LINKED ISSUE #147: Upload functionality must convert text file into PDF with extracted tokens
When user uploads `.txt` into BadgerDoc, this functionality must convert it into PDF and extract tokens

### PR DESCRIPTION
**Pull request by thinklab**
closes #147

### CODE REVIEW

File: assets/.env [hunk 1]
```diff
@@ -33,3 +33,5 @@ GOTENBERG_LIBRE_OFFICE_ENDPOINT="http://${GOTENBERG}/forms/libreoffice/convert"
 GOTENBERG_FORMATS=[".txt",".docx",".doc",".bib",".xml",".csv",".fodt",".html",".ltx",".odt",".ott",".pdb",".psw",".rtf",".sdw",".stw",".sxw",".uot",".vor",".wps",".epub",".emf",".fodg",".met",".odd",".otg",".std",".svg",".svm",".swf",".sxd",".sxw",".tiff",".xhtml",".xpm",".fodp",".potm",".pot",".pptx",".pps",".ppt",".pwp",".sda",".sdd",".sti",".sxi",".uop",".wmf",".odp"]
```
Comment (serereg) [inline assets/.env:34]: It is better to move *,txt from GOTENBERG_FORMATS to new constant CONVERT_FORMATS

File: assets/src/utils/common_utils.py [hunk 2]
```diff
@@ -212,11 +218,40 @@ def convert_to_jpg(self) -> bytes:
...
+        return converted_file.get(output_pdf_path).read()
+
     def convert(self) -> Union[bool]:
         """
         Checks if file format is in the available conversion formats.
         """
         try:
+            if self.ext == "txt":
+                self.converted_file = self.convert_txt()
             if self.ext in settings.gotenberg_formats:
```
Comment (serereg) [inline assets/src/utils/common_utils.py:255]: txt should be excluded from gotenberg

**Review by serereg — COMMENTED** [review]
From my point of view there should be changes in deployment.yaml

File: assets/.env [hunk 3]
```diff
@@ -33,3 +33,5 @@ GOTENBERG_LIBRE_OFFICE_ENDPOINT="http://${GOTENBERG}/forms/libreoffice/convert"
...
 
 IMAGE_FORMATS=[".png",".bmp", ".pbm", ".pct", ".pgm", ".ppm", ".ras", ".tiff"]
+
+SERVICE_CONVERT_URI="http://convert/text/import"
```
Comment (serereg) [inline assets/.env:37]: From my point of view this variable should be also defined in deployment.yaml

File: assets/src/utils/common_utils.py [hunk 4]
```diff
@@ -212,11 +218,40 @@ def convert_to_jpg(self) -> bytes:
         self.conversion_status = "converted to JPEG"
         return byte_im
 
+    def convert_txt(self):
+        input_text_path = (
+            f"files/{self.new_file.id}/"
+            f"{self.new_file.id}{self.new_file.original_ext}"
+        )
+        output_pdf_path = f"files/{self.new_file.id}/{self.new_file.id}.pdf"
+        output_tokens_path = f"files/{self.new_file.id}/tokens/1.json"
```
Comment (serereg) [inline assets/src/utils/common_utils.py:227]: rename tokens -> ocr

**Review by khyurri — APPROVED** [review]

File: convert/Dockerfile [hunk 5]
```diff
@@ -1,4 +1,4 @@
-ARG base_image=818863528939.dkr.ecr.eu-central-1.amazonaws.com/badgerdoc/python_base:0.1.7
```
Comment (serereg) [inline convert/Dockerfile:1]: Remove it please

File: assets/Dockerfile [hunk 6]
```diff
@@ -1,4 +1,4 @@
-ARG base_image=818863528939.dkr.ecr.eu-central-1.amazonaws.com/badgerdoc/python_base:0.1.7
+ARG base_image=818863528939.dkr.ecr.eu-central-1.amazonaws.com/badgerdoc/python_base:0.1.6
```
Comment (serereg) [inline assets/Dockerfile:1]: Remove it please

**Pull request merged**